File: 8KYU/Tip Calculator/solution.cs
using System;
public class Kata
{
	public static int CalculateTip(double amount, string rating)
	{
		switch(rating.ToLower())
		{
		  case "excellent":
			return (int)Math.Ceiling(amount * 0.20D);
		  case "great":
			return (int)Math.Ceiling(amount * 0.15D);
		  case "good":
			return (int)Math.Ceiling(amount * 0.10D);
		  case "poor":
			return (int)Math.Ceiling(amount * 0.05D);
		  case "terrible":
			return 0;
		  default:
			return -1;
		}
	}
}

File: 8KYU/Tip Calculator/solution.py
import math
def calculate_tip(amount, rating):
    if rating.lower() == 'excellent':
        return math.ceil(amount * 0.20)
    elif rating.lower() == 'great':
        return math.ceil(amount * 0.15)
    elif rating.lower() == 'good':
        return math.ceil(amount * 0.10)
    elif rating.lower() == 'poor':
        return math.ceil(amount * 0.05)
    elif rating.lower() == 'terrible':
        return 0
    else:
        return "Rating not recognised"